src/main/java/org/opensearch/ad/cluster/ADDataMigrator.java

```diff
@@ -1,301 +1,301 @@
+
+import static org.opensearch.ad.constant.CommonName.DETECTION_STATE_INDEX;
+import static org.opensearch.ad.model.ADTask.DETECTOR_ID_FIELD;
+import static org.opensearch.ad.model.ADTask.IS_LATEST_FIELD;
+import static org.opensearch.ad.model.ADTask.TASK_TYPE_FIELD;
+import static org.opensearch.ad.model.ADTaskType.taskTypeToString;
+import static org.opensearch.ad.model.AnomalyDetector.ANOMALY_DETECTORS_INDEX;
+import static org.opensearch.ad.model.AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX;
+import static org.opensearch.ad.settings.AnomalyDetectorSettings.MAX_DETECTOR_UPPER_LIMIT;
+import static org.opensearch.ad.util.RestHandlerUtils.XCONTENT_WITH_TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.createXContentParserFromRegistry;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.ResourceAlreadyExistsException;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.ad.common.exception.ResourceNotFoundException;
+import org.opensearch.ad.constant.CommonName;
+import org.opensearch.ad.indices.AnomalyDetectionIndices;
+import org.opensearch.ad.model.ADTask;
+import org.opensearch.ad.model.ADTaskState;
+import org.opensearch.ad.model.ADTaskType;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.AnomalyDetectorJob;
+import org.opensearch.ad.model.DetectorInternalState;
+import org.opensearch.ad.rest.handler.AnomalyDetectorFunction;
+import org.opensearch.ad.util.ExceptionUtil;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.MatchAllQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.sdk.SDKClient;
+import org.opensearch.sdk.SDKClusterService;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+/**
+ * Migrate AD data to support backward compatibility.
+ * Currently we need to migrate:
+ * 1. Detector internal state (used to track realtime job error) to realtime data.
+ */
+public class ADDataMigrator {
+    private final Logger logger = LogManager.getLogger(this.getClass());
+    private final SDKClient.SDKRestClient client;
```

Comment:
sdkRestClient

Reply:
I would suggest to keep the consistency in the entire project. I think there is a diff then it will not be a major one. Also, it increases readability.

Reply:
I'll have to agree with @owaiskazi19 on this one, naming conventions aside, minimizing the diff will aid in overall migration efforts and will save us some headache later down the line once we rebase with the main branch.